doc/developer/design/20250226_postgres_style_explain.md

```diff
@@ -0,0 +1,338 @@
+syntax.
+
+## Out of Scope
+
+We are not going to build new `EXPLAIN` infrastructure, diagrams,
+etc. For example, we are not going to attempt to differentiate between
+the different meanings of `ArrangeBy` in MIR.
+
+We are not going to invent fundamentally new ways of explaining
+how Materialize works.
+
+We are not going to do a user study in advance of any changes. (But we
+will listen attentively to feedback!)
+
+## Solution Proposal
+
+Postgres explain plans have the format:
+
+```
+Operator
+  Detail
+  -> Child Operator #1
+     Detail
+     ...
+  -> Child Operator #2
+     Detail
+     ...
+```
+
+We should aim to follow Postgres's norms: operator names spelled out
+with spaces, and properties are clearly elucidated in human-readable
+formats. When it is sensible, we have simply borrowed Postgres's
+terminology, i.e., `Reduce` is renamed to `GroupAggregate`.
+
+Postgres displays some parts of the query differently from us, namely:
+
+  - Column names:
+    + When a column name is available, it just gives the name (no number).
+    + When a column name is unavailable, it gives the number using `$2`.
+  - `Map` and `Project` do not appear
+
+We will use LIR as the new default `EXPLAIN`/`EXPLAIN AS TEXT`
+output. We will update `mz_lir_mapping` to use the new Postgres-style
+syntax (fixing [a bug with `Let` and `LetRec`
+rendering](https://github.com/MaterializeInc/database-issues/issues/8993)
+in the process.
+
+We will need three pieces of work, which should all land together:
+
+ - We must implement the new output and put in appropriate SLT tests
+   for it.
+ - We must update `mz_lir_mapping` to use the new vocabulary.
+ - We must update the documentation to explain the new output, ideally
+   using this output everywhere `EXPLAIN` is used.
+
+### Concrete Mapping
+
+| LIR node    | `mz_lir_mapping` node                    | New, Postgres-style syntax              |
+| :---------- | :--------------------------------------- | :-------------------------------------- |
+| `Constant`  | `Constant`                               | `Constant`                              |
+| `Get`       | `Get::PassArrangements l0`               | `Index Scan on l0 using ...`            |
+| `Get`       | `Get::Arrangement l0 (val=...)`          | `Index Lookup on l0 using ...`          |
+| `Get`       | `Get::Collection l0`                     | `Read l0`                               |
+| `Mfp`       | `MapFilterProject`                       | `Map/Filter/Project`                    |
+| `FlatMap`   | `FlatMap`                                | `Flat Map`                              |
+| `Join`      | `Join::Differential`                     | `Differential Join`                     |
+| `Join`      | `Join::Delta`                            | `Delta Join`                            |
+| `Reduce`    | `Reduce::Distinct`                       | `Distinct GroupAggregate`               |
```

Comment:
Why not just `Distinct`?

Reply:
Just following Postgres, which always calls reduces a `GroupAggregate`:

```
EXPLAIN SELECT DISTINCT l_discount FROM lineitem;
                            QUERY PLAN                            
------------------------------------------------------------------
 HashAggregate  (cost=12.50..14.50 rows=200 width=18)
   Group Key: l_discount
   ->  Seq Scan on lineitem  (cost=0.00..12.00 rows=200 width=18)
(3 rows)
```

Reply:
I think it's fine to show it as `Distinct`. I'd say this is a clear improvement over Postgres not showing it as `Distinct`. There is no need to follow Postgres just for familiarity here, because we don't really need to rely on familiarity, as it should be pretty clear for users what `Distinct` is.